test/sequence/pfr_fields_view.cpp

```diff
@@ -0,0 +1,151 @@
+/*=============================================================================
+    Copyright (c) 2022 Denis Mikhailov
+    Distributed under the Boost Software License, Version 1.0. (See accompanying
+    file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+==============================================================================*/
+#include <boost/detail/lightweight_test.hpp>
+#include <boost/fusion/view/pfr_fields_view.hpp>
+#include <boost/fusion/sequence/intrinsic/at.hpp>
+#include <boost/fusion/sequence/intrinsic/size.hpp>
+#include <boost/fusion/sequence/intrinsic/empty.hpp>
+#include <boost/fusion/sequence/intrinsic/front.hpp>
+#include <boost/fusion/sequence/intrinsic/back.hpp>
+#include <boost/fusion/support/is_view.hpp>
+#include <boost/fusion/sequence/io/out.hpp>
+#include <boost/fusion/sequence/comparison/equal_to.hpp>
+#include <boost/fusion/sequence/comparison/not_equal_to.hpp>
+#include <boost/fusion/sequence/comparison/less.hpp>
+#include <boost/fusion/sequence/comparison/less_equal.hpp>
+#include <boost/fusion/sequence/comparison/greater.hpp>
+#include <boost/fusion/sequence/comparison/greater_equal.hpp>
+#include <boost/fusion/container/vector/vector.hpp>
+#include <boost/fusion/container/list/list.hpp>
+#include <boost/fusion/container/generation/make_vector.hpp>
+#include <boost/fusion/mpl.hpp>
+#include <boost/mpl/front.hpp>
+#include <boost/mpl/is_sequence.hpp>
+#include <boost/mpl/assert.hpp>
+#include <iostream>
+#include <string>
+
+namespace namespaced_type {
+    typedef int integer;
+}
+
+namespace ns {
+    struct point {
+        int x;
+        int y;
+        namespaced_type::integer z;
+    };
+
+    // Testing non-constexpr compatible types
+#if BOOST_PFR_USE_CPP17 != 0
+    struct employee {
+        std::string name;
+        std::string nickname;
+    };
+#endif // BOOST_PFR_USE_CPP17
```

Comment:
Does it mean we loose the non-constexpr compatible types testcase on C++17 compilation ?

Reply:
In C++17 we can't create constexpr `std::string`. Did you mean to say in C++20?
If I understood you correctly, this structure should be brought to like this form:
```
struct employee {
    guaranteed_nonconstexpr_string name;
    guaranteed_nonconstexpr_string nickname;
};
```

Reply:
Oh yes I mixed it with C++20 that has constexpr allocator and std::string.

So I'm unsure why you disable it on C++17 and if this is the purpose of the test to test non constexpr cases (which makes sense) I would definitely add what you propose with guaranteed_nonconstexpr_string.

Reply:
> So I'm unsure why you disable it on C++17 

Real? I suppose I disable it on C++17 and enable on C++14.

> I would definitely add what you propose with guaranteed_nonconstexpr_string.

Ok, I will add `guaranteed_nonconstexpr_string`

Reply:
PS typo fixed